### DISCUSSION TITLE
코드 하이라이팅 색상 변경

### DISCUSSION THREAD

**Discussion by ImSejin** [pronist/hELLO#221] [Q&A] [answered]
> 코드 하이라이트는 라이트모드/다크모드 분리에 따른 하이라이트 테마 두 개를 요구하는 관계로 스킨에 내장되어 있습니다. 
> **티스토리 플러그인 - 코드 문법 강조**에 해당하는 플러그인을 반드시 꺼주시기 바랍니다.

말씀하신 대로 티스토리에서 제공하는 플러그인을 껐는데, 생각만큼 하이라이팅이 잘 안 됩니다.

### 보기 화면과 수정 화면의 하이라이팅이 달라요.

보기 화면과 수정 화면의 하이라이팅이 다르던데, 보기 화면의 하이라이팅을 수정 화면과 동일하게 적용할 수 있나요?

![image](https://github.com/pronist/tistory-skin-hELLO/assets/46176032/cd776164-6e3c-4d3a-8650-df77519a7a96)
보기 화면

![image](https://github.com/pronist/tistory-skin-hELLO/assets/46176032/4596fcd3-5d57-44f5-9b1a-61e19674844e)
수정 화면

### 하이라이팅 색상을 커스터마이징하고 싶어요.

하이라이팅 색상을 직접 변경하는 방법은 `index.xml`에서 default를 수정하는 방법 말곤 없나요? 커스텀 테마를 적용하고 싶어요.

```xml
 <variable>
   <name>hljs</name>
   <label>코드 하이라이팅(라이트)</label>
   <description><![CDATA[ 코드 하이라이팅(라이트) ]]></description>
   <type>STRING</type>
   <default>xcode</default>
 </variable>
 <variable>
   <name>hljs-dark</name>
   <label>코드 하이라이팅(다크)</label>
   <description><![CDATA[ 코드 하이라이팅(다크) ]]></description>
   <type>STRING</type>
   <default>vs2015</default>
 </variable>
```

**Comment by pronist** [comment, marked as answer]
@ImSejin 안녕하세요. 제가 이 시기에 개발활동을 안 했어서 답변은 이제 드리는 점 양해 부탁드립니다. 

1. 에디터의 화면과 스킨에 출력되는 코드 테마가 다른 것은 에디터의 경우 티스토리에서 지정한 부분이므로 스킨 개발자가 어찌할 수 없고, 건드릴 수 있는 부분은 스킨에 출력되는 부분이라 서로 다릅니다.
2. 코드 하이라이팅의 경우 `index.xml` 뿐만 아니라 **관리자 - 스킨 편집**에서 옵션을 조절할 수 있는 부분에서 테마를 설정할 수 있습니다. 단, 현재 스킨에 설정된 버전이 코드 하이라이팅이 11.7.0 이므로 <https://github.com/highlightjs/highlight.js/tree/11.7.0/src/styles> 에서 제공되는 테마에 한해 설정할 수 있을 것입니다.

![image](https://github.com/pronist/hello/assets/25624965/b860f0af-f5bc-4ba0-9fa9-f544eb4443a0)
